codec/src/main/java/io/netty/handler/codec/AsciiString.java

```diff
@@ -1090,17 +783,24 @@ public boolean regionMatches(boolean ignoreCase, int thisStart, CharSequence str
      * @return a new string with occurrences of oldChar replaced by newChar.
      */
     public AsciiString replace(char oldChar, char newChar) {
-        int index = indexOf(oldChar, 0);
+        final int index;
+        try {
+            index = forEachByte(new IndexOfVisitor((byte) oldChar));
```

Comment:
We could return `this` simply when `oldChar > 255` and assume `oldChar` is always <= 255. Also, I would do `byte newCharByte = c2b(newChar)` from the beginning.